File: Clients/ExchangeRates.Client/ExchangeRates.Blazor.Client/Models/ExchangeRateResult.cs
namespace ExchangeRates.Blazor.Client.Models;

public class ExchangeRateResult {
	public DateOnly Date { get; set; }
	public string BaseCurrency { get; set; } = string.Empty;
	public string QuoteCurrency { get; set; } = string.Empty;
	public decimal Rate { get; set; }
	public string Provider { get; set; } = string.Empty;
}

File: Clients/ExchangeRates.Client/exchangerates.client.react/src/models/ExchangeRateResult.js
export default class ExchangeRateResult {
    constructor({ date, baseCurrency, quoteCurrency, rate, provider }) {
        this.date = String(date);
        this.baseCurrency = String(baseCurrency);
        this.quoteCurrency = String(quoteCurrency);
        this.rate = Number(rate);
        this.provider = String(provider);
    }
}